ios/RCTPSPDFKit/RCTPSPDFKitView.m

```diff
@@ -203,6 +203,37 @@ - (void)addAnnotation:(id)jsonAnnotation {
   }
 }
 
+- (void)removeAnnotation:(id)jsonAnnotation {
+  NSData *data;
+  if ([jsonAnnotation isKindOfClass:NSString.class]) {
+    data = [jsonAnnotation dataUsingEncoding:NSUTF8StringEncoding];
+  } else if ([jsonAnnotation isKindOfClass:NSDictionary.class])  {
+    data = [NSJSONSerialization dataWithJSONObject:jsonAnnotation options:0 error:nil];
+  } else {
+    NSLog(@"Invalid JSON Annotation.");
+    return;
+  }
+
+  PSPDFDocument *document = self.pdfController.document;
+  PSPDFDocumentProvider *documentProvider = document.documentProviders.firstObject;
+
+  BOOL success = NO;
+  if (data) {
+    PSPDFAnnotation *annotationToRemove = [PSPDFAnnotation annotationFromInstantJSON:data documentProvider:documentProvider error:NULL];
+    for (PSPDFAnnotation *annotation in [document annotationsForPageAtIndex:annotationToRemove.pageIndex type:annotationToRemove.type]) {
+      // Remove the annotation if the name matches.
+      if ([annotation.name isEqualToString:annotationToRemove.name]) {
+        success = [document removeAnnotations:@[annotation] options:nil];
+        break;
+      }
+    }
+  }
+
+  if (!success) {
+    NSLog(@"Failed to remove annotation.");
```

Comment:
I think instead of just logging the error we should report the result back to react so users can actually handle it. I think just a boolean indicating if the annotation was removed is enough.